components/support/migration/src/main/java/mozilla/components/support/migration/GeckoMigration.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import java.io.File
+import java.io.IOException
+
+private const val PREFS_FILE = "prefs.js"
+
+/**
+ * Helper for migrating Gecko related internal files.
+ */
+internal object GeckoMigration {
+    /**
+     * Tries to migrate Gecko related internal files in [profilePath].
+     */
+    fun migrate(
+        profilePath: String
+    ): Result<Unit> {
+        // The prefs file contains gecko prefs that have been changed by the user (about:config) or
```

Comment:
Maybe also worth noting in a comment here that GeckoView will just happily inherit whatever is there, so our "migration" really is just about removing that prefs file.